doc/plugin-criteria.md

```diff
@@ -0,0 +1,51 @@
+|----|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| 1. |  The task is already solved by kubectl.                                                                                                                                                                                        |  Mere command wrappers are not acceptable, as those are better covered by shell aliases.  However, convenience and productivity improvements are welcome (examples: `konfig`, `change-ns`, `view-secret`) |
+| 2. | The task provides a different solution to a problem which is also solved by kubectl. However, the task would be impossible or very hard to reproduce with kubectl alone. Examples: `access-matrix`, `get-all`, `mtail`, `tail` |  In general welcome unless it overlaps significantly with other plugins or has a bad usability.                                                                                                           |
+| 3. | The task addresses a completely different problem than any of the kubectl subcommands, but is clearly focused on k8s.                                                                                                          |  Gray area. If the task fits well into `kubectl`, it may be accepted (for example `kubectl debug`).                                                                                                       |
+| 4. | The task is very dissimilar to common kubectl tasks and is not primarily focused on k8s. However, it can be useful in the k8s ecosystem.                                                                                       | Not acceptable.                                                                                                                                                                                           |
+| 5. | The task is very dissimilar to common kubectl tasks and is not focused on k8s. Its general usefulness in the k8s ecosystem is questionable.                                                                                    | Not acceptable.                                                                                                                                                                                           |
```

Comment:
Being arbiters if utility is bound to cause some friction. Usefulness is in the eye of the beholder. Homebrew hacks around this by saying packages have to be maintained by users, not by the original author. The gist being if a user is happy to maintain a package them by definition it's useful. Not saying Krew should follow the same pattern, but having guidelines which say "we decide based on what we think" might cause friction later.

Reply:
I agree that "usefulness" is rather subjective. But I think the krew index does need some form of filtering. Otherwise, where should the line be drawn? If there is no line (like for brew), krew would be come a mere package manager and not a kubectl package manager.
Maybe you have better ideas of what could be reasonable criteria?